vertical-pod-autoscaler/pkg/admission-controller/logic/server.go

```diff
@@ -121,11 +121,18 @@ func (s *AdmissionServer) getPatchesForPodResourceRequest(raw []byte, namespace
 		updatesAnnotation = append(updatesAnnotation, fmt.Sprintf("container %d: ", i)+strings.Join(annotations, ", "))
 	}
 	if len(updatesAnnotation) > 0 {
-		patches = append(patches, patchRecord{
-			Op:   "add",
-			Path: "/metadata/annotations",
-			Value: map[string]string{
-				"vpaUpdates": fmt.Sprintf("Pod resources updated by %s: ", vpaName) + strings.Join(updatesAnnotation, "; ")}})
+		if pod.Annotations == nil {
+			patches = append(patches, patchRecord{
+				Op:   "add",
+				Path: "/metadata/annotations",
+				Value: map[string]string{
+					"vpaUpdates": fmt.Sprintf("Pod resources updated by %s: ", vpaName) + strings.Join(updatesAnnotation, "; ")}})
```

Comment:
Can you construct the annotation value string in one place? can be a variable or function.